### PR TITLE
Add Telemetry Manager module prow jobs

### PR DESCRIPTION
**Pull request by hisarbalik**
**Description**

Changes proposed in this pull request:

- Create and configure prow jobs (pull-telemetry-manager-build, pull-telemetry-manager-lint, pull-telemetry-manager-unit-test) for Telemetry Manager module, 

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/16745

### CODE REVIEW

File: prow/jobs/test-infra/pjconfigtests.yaml [hunk 1]
```diff
@@ -981,6 +981,47 @@ presubmits: # runs on PRs
...
+              requests:
+                memory: 1Gi
+                cpu: 2
+        tolerations:
+          - key: dedicated
+            operator: Equal
+            value: high-cpu
+            effect: NoSchedule
+        nodeSelector:
+            dedicated: "high-cpu"
```
Comment (skhalash) [inline prow/jobs/test-infra/pjconfigtests.yaml:1023]: Could you please explain what this job is doing?

**Review by skhalash — APPROVED** [review]

File: prow/jobs/test-infra/pjconfigtests.yaml (hunk 1 above)
Comment (hisarbalik) [reply prow/jobs/test-infra/pjconfigtests.yaml:1023]: I don't know, it is generated and could not found any info

**Review by Ressetkk — APPROVED** [review]

**Review by Ressetkk — APPROVED** [review]

**Pull request merged**